libs/custom-paper-handler/src/driver/driver.ts

```diff
@@ -477,53 +443,95 @@ export class PaperHandlerDriver {
     await this.transferOutGeneric(ScanCommand, undefined);
     debug('STARTING SCAN');
     let scanStatus = OK_CONTINUE;
+    let dataBlockBytesReceived = 0;
+    let width = -1;
     const imageData: Uint8Array[] = [];
 
+    // Assumptions:
+    // 1. There's at least 1 data block
+    // 2. Each data block can fit within a single `transferInGeneric` call
     while (scanStatus === OK_CONTINUE) {
       const rawResponse = await this.transferInGeneric();
       assert(rawResponse?.data);
 
       const responseBuffer = rawResponse.data.buffer;
-
       const header = responseBuffer.slice(0, SCAN_HEADER_LENGTH_BYTES);
       const response: ScanResponse = ScanResponse.decode(
         Buffer.from(header)
       ).unsafeUnwrap();
-
       scanStatus = response.returnCode;
       const { sizeX, sizeY } = response;
-      const pixelsPerByte = response.scan <= 5 ? 1 : 8;
+      // The last sizeX returned by the scan command is 0, so store the first one for use later
```

Comment:
I don't think I understand this comment. Where is the check for 0?

Reply:
We want to use `sizeX` returned by the scan command later in this function. Each packet sent by the scanner includes a header that specifies sizeX and sizeY. So for packets 0..n-1 we get a packet like this (pseudocode):

```
scanStatus: OK_CONTINUE,
sizeX: 1738,
sizeY: ...
dataBlock: // Scan image data
```

the last packet _n_ is empty so we get

```
scanStatus: OK_NO_MORE_DATA,
sizeX: 0,
sizeY: 0,
dataBlock: // no bytes
```

Ideally there would be no empty packet, and the last packet would be:

```
scanStatus: OK_NO_MORE_DATA,
sizeX: 1738,
sizeY: ...
dataBlock: // Scan image data
```

In that case we could just store the last `sizeX` value. But since we _do_ get an empty packet with `sizeX=0` we store `sizeX` from the first packet instead.

I wrote this comment because I felt the empty packet at the end was nonintuitive and it wasn't clear why we store the first `sizeX` value. I'll try to rewrite this comment to be more clear.